pkg/checks/checks.go

```diff
@@ -54,68 +56,182 @@ func WaitForHTTPStatusOk(ctx context.Context, transport http.RoundTripper, urlto
 	}, ctx.Done())
 }
 
-func checkNamespace(namespace string) bool {
-	if namespace == "default" {
-		return true
-	}
+type Node struct {
+	Ready bool
+	Name  string
+}
 
-	whitelistedNamespaces := []string{
-		"openshift-",
-		"kube-",
-	}
-	for _, ns := range whitelistedNamespaces {
-		if strings.HasPrefix(namespace, ns) {
-			return true
+func GenerateNodeMap(cs *acsapi.OpenShiftManagedCluster) map[string][]Node {
+	nodes := make(map[string][]Node)
+	for _, app := range cs.Properties.AgentPoolProfiles {
+		nodes[string(app.Role)] = []Node{}
+		for i := 0; i < app.Count; i++ {
+			n := Node{Name: fmt.Sprintf("%s-%06d", app.Role, i)}
+			nodes[string(app.Role)] = append(nodes[string(app.Role)], n)
 		}
 	}
-	return false
+	return nodes
 }
 
-// WaitForInfraServices verify daemonsets, statefulsets
-func WaitForInfraServices(ctx context.Context, appclient *appclient.AppsV1Client) error {
-	log.Info("checking infrastructure service health")
-out:
-	for {
-		_, err := appclient.Deployments("openshift-web-console").Get("webconsole", metav1.GetOptions{})
-		switch {
-		case err == nil:
-			break out
-		case kerrors.IsNotFound(err):
-		default:
-			return err
-		}
-		select {
-		case <-time.After(2 * time.Second):
-		case <-ctx.Done():
-			return ctx.Err()
+func WaitForNodeReady(ctx context.Context, nodes map[string][]Node, kc *kubernetes.Clientset) error {
```

Comment:
Instead of using `GenerateNodeMap`, it would be more succint to loop through the agentpoolprofiles here and list nodes as you go.

Reply:
I believe it would be more succinct.  Doing it this way for a couple of reasons.
1. Because Jim recommended it: https://github.com/openshift/openshift-azure/pull/328#pullrequestreview-152724916
2. This is a way to validate that the node map is valid.  This map is then used later on to figure out which static pods belong on each host.